Server/ConcordiaCurriculumManager/Services/CourseGroupingService.cs

```diff
@@ -222,4 +223,33 @@ public async Task DeleteCourseGroupingRequest(Guid dossierId, Guid requestId)
             throw new ServiceUnavailableException("The course grouping could not be deleted");
         }
     }
+
+    public async Task<CourseGrouping> PublishCourseGrouping(Guid commonIdentifier)
+    {
+        var newCourseGrouping = await _courseGroupingRepository.GetCourseGroupingByCommonIdentifierAnyState(commonIdentifier) ?? throw new NotFoundException($"The course grouping with common ID {commonIdentifier} was not found.");
+        var oldCourseGrouping = await _courseGroupingRepository.GetPublishedVersion(commonIdentifier);
```

Comment:
You'll need to either `QueryRelatedCourseGroupingData` for oldCourseGroupings or remove the call to `VerifyCourseGroupingIsValidOrThrow()` when unpublishing the grouping, because it'll crash as is now.

Reply:
Huh, it didn't crash for me. But I'll remove `VerifyCourseGroupingIsValidOrThrow()` just in case,

Reply:
Yes sorry you're right, it wouldn't crash because SubGroupings and Courses are initialized to an empty list when not loaded, they are never null (but as an empty list still doesn't quite work correctly anyways)